doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "PARTITION BY"
```

Comment:
This doc covers both the `PARTITION BY` syntax and the filter pushdown optimization, which are strictly speaking two separate things but likely to be used together I think?

Reply:
👍 Works for me.
We could title it PARTITION BY/filter pushdown optimization or Filter pushdown optimzation (PARTITION BY) -- not quite sure the naming scheme of the pages in the patterns section.

Reply:
Oh, should we at least mention add a link to this page in the temporal filters pattern page?

Reply:
Great - took another stab at the name. (The all-uppercase version seemed out of step with the other guides.)

It does make sense to link from that other doc, but I had a hard time finding a way to do it that felt appropriate for a private preview feature. I left a TODO for now, but I'm happy for other suggestions.

Reply:
❤️ the name.
As for the temporal filters page, maybe add a "See also [Partitioning and filter pushdown](/transform-data/patterns/partition-by/)" at the end of the page -- if we want to get more users to turn on the feature for them.